pkg/util/pullsecret/pullsecret_test.go

```diff
@@ -373,3 +376,79 @@ func TestUnmarshalSecretData(t *testing.T) {
 		})
 	}
 }
+
+func TestValidatePullSecret(t *testing.T) {
+	azurecrError := "unable to retrieve auth token: invalid username/password: unauthorized: authentication required, visit https://aka.ms/acr/authorization for more information. This error has been customized to ensure it doesn't leak to Azure."
+	erroringRegistry := RegistryClient{
+		CheckAuth: func(ctx context.Context, sc *types.SystemContext, s1, s2, s3 string) error {
+			return fmt.Errorf(azurecrError)
+		},
+	}
+	succeedingRegistry := RegistryClient{
+		CheckAuth: func(ctx context.Context, sc *types.SystemContext, s1, s2, s3 string) error {
+			return nil
+		},
+	}
+	test := []struct {
+		name     string
+		ps       *corev1.Secret
+		wantAuth map[string]string
+		wantErr  string
+		client   RegistryClient
+	}{
```

Comment:
If we're going to validate all tokens in the secret then we'll need a test case for a valid ACR token being checked after an invalid user token. I don't remember if go slices are guaranteed to be handled in order in a `for` loop but it would be good at least attempt it.